R/detect_column.r

```diff
@@ -0,0 +1,354 @@
+library(rio)
+library(tidyverse)
+library(DescTools)
+library(xts)
+
+#' Check if required columns exist, give an error if they do not exist.
+#' 
+#' @param data The dataset whose columns to check.
+#' @param domain_spec The domain specification ("CO", "DM", "SE", "SM", "SV") for checking columns.
+#' @return If required columns are missing, give an error and indicate which required columns are missing.
+#' @export
+validate_required_column_names <- function(data, domain_spec) {
```

Comment:
Can you please combine the `validate_required_column_names()`, `validate_expected_column_names()`, `validate_permitted_column_names()`, and `validate_extra_column_names()` into a single `validate_column_names()` function that tests everything?